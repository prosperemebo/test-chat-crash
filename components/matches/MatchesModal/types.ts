import { Banner, MatchFilters } from '@/types'

export interface MatchesModalProps {
  onClose: () => void
  setAgeRangeFilter: (value: [number, number]) => void
  setLanguagesFilter: (value: string[]) => void
  setGamesFilter: (value: Banner[]) => void
  setGenderFilter: (value: string) => void
  matchFilters: MatchFilters
  userGender: string | undefined
  updateFilters: () => void
  resetFilters: () => void
}

export interface GamesSelectorProps {
  isVisible: boolean
  onClose: () => void
  selectedGames: Banner[]
  onGamesChange: (game: Banner[]) => void
  isGamesRequestRunning?: boolean
  onboardingGames?: any[]
  getInitialOnboardingGames?: () => Promise<void>
  loadMoreOnboardingGames?: () => void
  searchGames?: (query: string) => Promise<void>
  setSelectedGames?: (games: Banner[]) => Promise<void>
}