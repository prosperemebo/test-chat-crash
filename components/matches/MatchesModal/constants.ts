import { Dimensions } from 'react-native'

export const DRAG_THRESHOLD = 50
export const GENDER_OPTIONS = ['Male', 'Female', 'Other']
export const SCREEN_WIDTH = Dimensions.get('window').width