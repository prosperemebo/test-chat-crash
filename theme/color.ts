export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}