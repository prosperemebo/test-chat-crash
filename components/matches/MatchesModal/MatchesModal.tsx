import React, { useState } from 'react'
import { TouchableOpacity } from 'react-native'
import MultiSlider from '@ptomasroos/react-native-multi-slider'
import * as Haptics from 'expo-haptics'
import {
  ModalContainer,
  Container,
  Header,
  CancelButton,
  Title,
  Label,
  GenderContainer,
  ButtonMatchesContainer,
  ResetButton,
  ResetText,
  ApplyButton,
  ApplyText,
  DragIndicator,
  FilterContainer,
  DragArea,
  ContentContainer,
  AgeHeaderContainer,
  AgeText,
  DragAreaContainer,
  SelectedGamesButton,
  SelectedGamesText,
} from './styles'
import CustomDropdown from '@/components/common/CustomDropdown/CustomDropdown'
import { useMarkerAnimation } from './hooks/useMarkerAnimation'
import { useModalDrag } from './hooks/useModalDrag'
import { CustomMarker } from './components/CustomMarker'
import { GenderButton } from './components/GenderButton'
import i18n from '@/localization/i18n'
import { useTheme } from 'styled-components/native'
import { locales } from '@/constants/locales'
import { MatchesModalProps } from './types'
import { DRAG_THRESHOLD, GENDER_OPTIONS, SCREEN_WIDTH } from './constants'
import GamesSelector from './components/GamesSelector'
import Icon from '@expo/vector-icons/Ionicons'
import compose from 'lodash/flowRight'
import { inject } from 'mobx-react'
import { Banner, Store } from '@/types'
import analytics from '@/services/analytics'

const MatchesModal = ({
  onClose, setAgeRangeFilter, setLanguagesFilter,
  setGamesFilter, setGenderFilter, matchFilters,
  updateFilters, resetFilters,
}: MatchesModalProps) => {
  const theme = useTheme()
  const [isGamesSelectorVisible, setGamesSelectorVisible] = useState(false)

  const { isDragging, markerScale, handleMarkerPress, handleMarkerRelease } = useMarkerAnimation()
  const { pan, panResponder } = useModalDrag(onClose, DRAG_THRESHOLD)

  const handleValuesChange = (values: number[]) => {
    setAgeRangeFilter([values[0], values[1]])

    if (values[0] % 5 === 0 || values[1] % 5 === 0) {
      Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light)
    }
  }

  const handleApply = () => {
    analytics.matches.matchesFilterApplied(matchFilters)
    updateFilters()
    onClose()
  }

  const handleReset = () => {
    analytics.matches.matchesFilterReset()
    resetFilters()
  }

  const languageOptions = locales.map(locale => ({
    label: locale.toUpperCase(),
    value: locale,
  }))

  const handleGamesChange = (games: Banner[]) => {
    setGamesFilter(games)
  }

  return (
    <ModalContainer>
      <Container style={{ transform: [{ translateY: pan.y }] }}>
        <DragAreaContainer>
          <DragArea {...panResponder.panHandlers}>
            <DragIndicator />
          </DragArea>
        </DragAreaContainer>
        <ContentContainer>
          <Header>
            <Title>{i18n.t('matches:filterMatches')}</Title>
            <TouchableOpacity onPress={onClose}>
              <CancelButton>{i18n.t('common:cancel')}</CancelButton>
            </TouchableOpacity>
          </Header>
          <FilterContainer>
            <>
              <Label>{i18n.t('onb:gender')}</Label>
              <GenderContainer>
                {GENDER_OPTIONS.map(option => (
                  <GenderButton
                    key={option}
                    option={option}
                    active={matchFilters.gender === option}
                    onPress={() => setGenderFilter(option)}
                  />
                ))}
              </GenderContainer>
            </>

            <AgeHeaderContainer>
              <Label>{i18n.t('common:ageRange')}</Label>
              <AgeText>{`${matchFilters.ageRange[0]} ~ ${matchFilters.ageRange[1]}`}</AgeText>
            </AgeHeaderContainer>

            <MultiSlider
              values={matchFilters.ageRange}
              min={13}
              max={90}
              containerStyle={{
                width: '100%',
                paddingHorizontal: 10,
                alignSelf: 'center',
              }}
              onValuesChange={handleValuesChange}
              step={1}
              selectedStyle={{
                backgroundColor: theme.common.gradientButtonColorStart,
                opacity: isDragging ? 0.8 : 1,
                height: 4,
              }}
              unselectedStyle={{
                backgroundColor: theme.common.buttonDartBackgroundColor,
                opacity: isDragging ? 0.6 : 1,
                height: 4,
              }}
              customMarker={() => <CustomMarker markerScale={markerScale} isDragging={isDragging} />}
              onValuesChangeStart={handleMarkerPress}
              onValuesChangeFinish={handleMarkerRelease}
              sliderLength={SCREEN_WIDTH - 55}
              trackStyle={{
                height: 4,
                borderRadius: 2,
              }}
            />

            <Label>{i18n.t('matches:selectGame')}</Label>
            <SelectedGamesButton onPress={() => setGamesSelectorVisible(true)}>
              <SelectedGamesText>
                {matchFilters.games.length > 0 ? matchFilters.games[0].title : i18n.t('matches:anyGame')}
              </SelectedGamesText>
              <Icon
                name="chevron-down"
                size={24}
                color={theme.common.inputTextColor}
              />
            </SelectedGamesButton>

            <GamesSelector
              isVisible={isGamesSelectorVisible}
              onClose={() => setGamesSelectorVisible(false)}
              selectedGames={matchFilters.games}
              onGamesChange={handleGamesChange}
            />

            <Label>{i18n.t('matches:selectLanguage')}</Label>
            <CustomDropdown
              data={languageOptions}
              value={matchFilters.languages}
              onChange={values => setLanguagesFilter(values as never[])}
              placeholder={i18n.t('matches:selectLanguage')}
            />

          </FilterContainer>
          <ButtonMatchesContainer>
            <ResetButton onPress={handleReset}>
              <ResetText>{i18n.t('common:reset')}</ResetText>
            </ResetButton>
            <ApplyButton onPress={handleApply}>
              <ApplyText>{i18n.t('matches:apply')}</ApplyText>
            </ApplyButton>
          </ButtonMatchesContainer>
        </ContentContainer>
      </Container>
    </ModalContainer>
  )
}

export default compose(
  inject((context: Store) => ({
    setAgeRangeFilter: context.MatchesStore.setAgeRangeFilter,
    setLanguagesFilter: context.MatchesStore.setLanguagesFilter,
    setGamesFilter: context.MatchesStore.setGamesFilter,
    setGenderFilter: context.MatchesStore.setGenderFilter,
    matchFilters: context.MatchesStore.matchFilters,
    userGender: context.UserStore.user?.profile.gender,
    updateFilters: context.MatchesStore.updateFilters,
    resetFilters: context.MatchesStore.resetFilters,
  })),
)(MatchesModal)