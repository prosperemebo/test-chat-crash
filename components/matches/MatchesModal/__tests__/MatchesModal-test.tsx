import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { Provider } from 'mobx-react'
import { describe, expect, test, jest } from '@jest/globals'
import MatchesModal from '../'
import { dark } from '@/theme'

const mockStore = {
  GamesStore: {
    isRequestRunning: false,
    onboardingGames: [],
    getInitialOnboardingGames: jest.fn(),
    loadMoreOnboardingGames: jest.fn(),
    searchGames: jest.fn(),
    setSelectedGames: jest.fn(),
  },
  MatchesStore: {
    matches: [],
    matchFilters: {
      ageRange: [13, 90],
      games: [],
      gender: 'Male',
      languages: [],
    },
    setGenderFilter: jest.fn(),
    setAgeRangeFilter: jest.fn(),
    setLanguageFilter: jest.fn(),
    setGamesFilter: jest.fn(),
    resetStore: jest.fn(),
  },
  UserStore: {
    user: {
      profile: {
        gender: 'Female',
      },
    },
  },
}

describe('<MatchesModal />', () => {
  test('MatchesModal renders correctly', () => {
    const tree = render(
      <Provider {...mockStore}>
        <ThemeProvider theme={dark}>
          <MatchesModal onClose={() => { }} />
        </ThemeProvider>
      </Provider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})