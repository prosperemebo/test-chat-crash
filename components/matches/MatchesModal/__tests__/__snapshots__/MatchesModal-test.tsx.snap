// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchesModal /> MatchesModal renders correctly 1`] = `
<View
  style={
    {
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#15161A",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 0,
        "marginTop": "auto",
        "maxHeight": "85%",
        "overflow": "hidden",
        "paddingBottom": 20,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#383B40",
          "height": 25,
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": "10%",
            "zIndex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "#858995",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 4,
              "marginBottom": 16,
              "marginTop": 8,
              "width": 40,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#383B40",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 16,
            "paddingHorizontal": 12,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 22,
              "textAlign": "center",
            }
          }
        >
          MOCKED_STRING:matches:filterMatches
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#873FBB",
                "fontSize": 16,
              }
            }
          >
            MOCKED_STRING:common:cancel
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingHorizontal": 12,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginVertical": 6,
            }
          }
        >
          MOCKED_STRING:onb:gender
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginHorizontal": 20,
                "opacity": 1,
              }
            }
          >
            <View
              active={true}
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderColor": "#873FBB",
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "borderWidth": 1,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "transform": [
                    {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "#873FBB1A",
                    "borderBottomLeftRadius": 10,
                    "borderBottomRightRadius": 10,
                    "borderTopLeftRadius": 10,
                    "borderTopRightRadius": 10,
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <Text
                active={true}
                collapsable={false}
                style={
                  {
                    "color": "#873FBB",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 14,
                    "textTransform": "capitalize",
                  }
                }
              >
                MOCKED_STRING:'onb:Male
              </Text>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginHorizontal": 20,
                "opacity": 1,
              }
            }
          >
            <View
              active={false}
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderColor": "#6B6E79",
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "borderWidth": 1,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "transform": [
                    {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "#873FBB1A",
                    "borderBottomLeftRadius": 10,
                    "borderBottomRightRadius": 10,
                    "borderTopLeftRadius": 10,
                    "borderTopRightRadius": 10,
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <Text
                active={false}
                collapsable={false}
                style={
                  {
                    "color": "#6B6E79",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "textTransform": "capitalize",
                  }
                }
              >
                MOCKED_STRING:'onb:Female
              </Text>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginHorizontal": 20,
                "opacity": 1,
              }
            }
          >
            <View
              active={false}
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderColor": "#6B6E79",
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "borderWidth": 1,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "transform": [
                    {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "#873FBB1A",
                    "borderBottomLeftRadius": 10,
                    "borderBottomRightRadius": 10,
                    "borderTopLeftRadius": 10,
                    "borderTopRightRadius": 10,
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <Text
                active={false}
                collapsable={false}
                style={
                  {
                    "color": "#6B6E79",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "textTransform": "capitalize",
                  }
                }
              >
                MOCKED_STRING:'onb:Other
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "marginVertical": 6,
              }
            }
          >
            MOCKED_STRING:common:ageRange
          </Text>
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 16,
              }
            }
          >
            13 ~ 90
          </Text>
        </View>
        <View
          style={
            {
              "alignSelf": "center",
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
          testID="mock-multi-slider"
        >
          <View
            collapsable={false}
            isDragging={false}
            style={
              {
                "backgroundColor": "#873FBB",
                "borderBottomLeftRadius": 15,
                "borderBottomRightRadius": 15,
                "borderColor": "#873FBB",
                "borderTopLeftRadius": 15,
                "borderTopRightRadius": 15,
                "borderWidth": 1,
                "height": 30,
                "shadowColor": "#873FBB",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 10,
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginVertical": 6,
            }
          }
        >
          MOCKED_STRING:matches:selectGame
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#15161A",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderColor": "#85899599",
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 16,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 16,
              }
            }
          >
            MOCKED_STRING:matches:anyGame
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": undefined,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginVertical": 6,
            }
          }
        >
          MOCKED_STRING:matches:selectLanguage
        </Text>
        <View
          style={
            {
              "backgroundColor": "#15161A",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderColor": "#6B6E79",
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "borderWidth": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
              "width": "100%",
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              {
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessible={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "flex": 1,
                      "fontSize": 16,
                      "writingDirection": "ltr",
                    },
                    {
                      "color": "#858995",
                      "fontSize": 18,
                    },
                    {},
                  ]
                }
              >
                MOCKED_STRING:matches:selectLanguage
              </Text>
              <View
                collapsable={false}
                style={
                  {
                    "transform": [
                      {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      {
                        "color": "#873FBB",
                        "fontSize": 20,
                      },
                      {
                        "fontFamily": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderColor": "#85899599",
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "borderWidth": 1,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
              "width": "38%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 18,
              }
            }
          >
            MOCKED_STRING:common:reset
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FF6705",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
              "width": "38%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 18,
              }
            }
          >
            MOCKED_STRING:matches:apply
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
